src/device_dot.cc

```diff
@@ -44,7 +44,21 @@ void dot(
     blas::internal_set_device( queue.device() );
 
     // call low-level wrapper
+    #if defined( BLAS_HAVE_ONEMKL )
+    sycl::queue syclq = queue.stream();
+    // check how the result scalar was allocated
+    auto result_ptr_type = sycl::get_pointer_type( result, syclq.get_context() );
+    if ( result_ptr_type == sycl::usm::alloc::unknown ) {
```

Comment:
Instead of doing an expensive device_malloc, the queue has a small amount of workspace that we can use. Cf.
https://github.com/icl-utk-edu/lapackpp/blob/master/src/cuda/cuda_potrf.cc#L162-L164

Also, where would this result_dev ever get freed? By using the queue's workspace, the queue handles freeing it.

Reply:
I have switched to using the device workspace as you recommended.
The result_dev never got freed.... my error.